parsec/mca/device/cuda/device_cuda_module.c

```diff
@@ -2579,12 +2596,15 @@ parsec_cuda_kernel_scheduler( parsec_execution_stream_t *es,
      *   - rc == 0: there is no manager, and at the exit of the while, this thread
      *             made rc go from 0 to 1, so it is the new manager of the GPU and
      *             needs to deal with gpu_task
+     *             or
+     *             There is a manager, but it is waiting for the co_manager to release
+     *             new tasks, so we just append our task and don't become manager
      *   - rc > 0: there is a manager, and at the exit of the while, this thread has
      *             committed new work that the manager will need to do, but the work is
      *             not in the queue yet.
      */
     while(1) {
-        rc = gpu_device->mutex;
+        rc = rc1 = gpu_device->mutex;
```

Comment:
I don't like `rc1` for its lack of information. I suggest using a boolean variable like `was_first_co_manager` (you get the point :)) that is set to `true` if the CAS below succeeds and `rc == 1`.